nengo/__init__.py

```diff
@@ -26,11 +26,13 @@
 from .learning_rules import PES, BCM, Oja, Voja
 from .params import Default
 from .probe import Probe
+from .processes import (BrownNoise, FilteredNoise, PresentInput, Process,
+                        WhiteNoise, WhiteSignal)
```

Comment:
> Not sure how much sense it makes to have only parts of a module as core part

We do currently have only parts in the core, either intentionally or unintentionally. For example `nengo.synapses.Triangle`. I guess since we're only going to be having a limited set of things in this repository anyway (e.g. new neuron types will mostly go in user repositories or some "extras" repo), that's another reason to import everything in a module. I'm thinking particularly with `dists` though, that there are some I'd like to be easily accessible, but I'm worried adding them all will clutter up the core namespace. It is a good point, though, that users don't want to worry about what may be imported and what's not. Furthermore, if something isn't imported users might not realize it exists if they don't know e.g. only some synapses are imported.

One argument for importing less is that it might get confusing what is the purpose of some of the base objects. For example, is `nengo.Gaussian` a distribution, process, or neuron type?